### DISCUSSION TITLE
how to use doctr instead of tesseract

### DISCUSSION THREAD

**Discussion by minooei** [deepdoctection/deepdoctection#136] [Q&A]
Hi thanks for your work
I couldn't find any documentation on how to use doctr instead of tesseract

**Comment by JaMe76** [comment]
Hi, 

thank you for your question. 

You can find a sample for a [pipeline](https://huggingface.co/spaces/deepdoctection/deepdoctection/blob/main/app.py) that uses doctr instead of Tesseract in the Space demo that unfortunately doesn't not work but for a different reason.

You will need to build a pipeline by yourself but you can copy the code from `build_analyzer(cfg)` in `deepdoctection.analyzer.dd` and replace the part where Tesseract is invoked (line 193-198) with

```python
# word detector
det = dd.DoctrTextlineDetector()
  
# text recognizer
rec = dd.DoctrTextRecognizer()

d_layout_text = dd.ImageLayoutService(det, to_image=True, crop_image=True)

pipe_component_list.append(d_layout_text)
 
d_text = dd.TextExtractionService(rec, extract_from_roi="WORD")
pipe_component_list.append(d_text)
``` 

Feel free to ask if there are any further questions.

**Reply by yumikim381** [reply]
Could you please specify which exact text to replace with current code version?

**Reply by JaMe76** [reply]
With the latest release you don't need to replace any code. You can use DocTR just by changing the `conf_dd_one.yaml` config:

```
OCR:
  USE_TESSERACT: False
  USE_DOCTR: True
```
